python/cudf/cudf/dataframe/series.py

```diff
@@ -1129,7 +1080,7 @@ def dropna(self):
         if self.null_count == 0:
             return self
         data = self._column.dropna()
-        index = self.index.loc[~self.isna()]
+        index = as_index(self.index.to_series().loc[~self.isna()])
```

Comment:
This will end up boolean masking a default index as well unnecessarily doing essentially double the work. We should probably fix this in the near future.

Reply:
Addressed in #2196